### PR TITLE
feat: add support for remote file parsing

### PR DESCRIPTION
**Pull request by tylerslaton**
## Summary

This PR adds support for remote file parsing. This will mainly be a nice feature for the SDKs as consumers of them can now parse file remotely prior to executing them. Namely, the UI will use this functionality.

## How
This is done by exporting `loader.ContentFromURL` as a new public function. The `input` package then uses `loader.ContentFromURL` in a new `input.FromURL` function. This allows all of the same loading logic to apply that GPTScript runs do while also leveraging the same cache that they do. Finally, the `parse` subcommand calls `input.FromURL` if a file load fails.

### CODE REVIEW

File: pkg/input/input.go [hunk 1]
```diff
@@ -39,3 +41,7 @@ func FromFile(file string) (string, error) {
 
 	return "", nil
 }
+
+func FromURL(url string) (string, error) {
```
Comment (thedadams) [inline pkg/input/input.go:45]: It seems that this function isn't necessary and callers can just call `loader.ContentFromURL`.

File: pkg/cli/parse.go [hunk 2]
```diff
@@ -26,12 +26,23 @@ func locationName(l string) string {
...
+		err     error
+	)
+
+	// Attempt to read the file first, if that fails, try to load the URL. Finally,
+	// return an error if both fail.
+	content, err = input.FromFile(args[0])
 	if err != nil {
-		return err
+		log.Debugf("failed to read file %s (due to %v) attempting to load the URL...", args[0], err)
+		content, err = input.FromURL(args[0])
```
Comment (thedadams) [inline pkg/cli/parse.go:39]: It's possible to get `"", nil` here. This indicates that `args[0]` isn't a URL or GitHub repo.

I think it makes sense to return the original error in this case. What do you think?

File: pkg/input/input.go (hunk 1 above)
Comment (tylerslaton) [reply pkg/input/input.go:45]: I mainly wanted them all of these to exist in the same package. Thoughts on that?

File: pkg/cli/parse.go (hunk 2 above)
Comment (tylerslaton) [reply pkg/cli/parse.go:39]: I agree and will add that.

File: pkg/input/input.go (hunk 1 above)
Comment (thedadams) [reply pkg/input/input.go:45]: Not something I think is valuable, but feel free to keep it.

File: pkg/cli/parse.go [hunk 3]
```diff
@@ -26,12 +28,27 @@ func locationName(l string) string {
...
+	// return an error if both fail.
+	content, err = input.FromFile(args[0])
 	if err != nil {
-		return err
+		log.Debugf("failed to read file %s (due to %v) attempting to load the URL...", args[0], err)
+		content, err = loader.ContentFromURL(args[0])
+		if err != nil {
+			return err
+		}
+		// If the content is empty and there was no error, this is not a remote file. Return the file error.
```
Comment (thedadams) [inline pkg/cli/parse.go:45]: nit: comment is out of date.

**Review by thedadams — APPROVED** [review]

**Review by cjellick — APPROVED** [review]

**Pull request merged**